### ISSUE TITLE
Repeated directory-name in 'file' completion mode

### ISSUE THREAD

**Issue by manasthakur** [lifepillar/vim-mucomplete#34]
Thanks for the awesome plugin.

My completion-chain is the following:
```
let g:mucomplete#chains = {}
let g:mucompletechains.default = ['file', 'c-p']
```
The problem is that while completing paths, upon pressing `tab`, I get the next portion of the directory-name written twice into the buffer.
Thus, for the path `/home/manas`, if I type `/ho<tab>`, it expands to `home/home`. Then I need to delete the second `home`. Once I write `/home/ma<tab>`, it expands to `/home/manas/manas`, and so on. Note that things are okay if I don't type anything after `/` and press `tab`.

This works perfectly fine with the default `<C-X><C-F>` file completion of Vim.

_p.s._ `path` completion provided by `vim-mucomplete` is _not_ what I want.

**Comment by lifepillar** [comment]
Are you using the latest version? I cannot reproduce the problem using the minimal `vimrc` suggested in `:h mucomplete-troubleshooting` with your completion chain.

**Comment by manasthakur** [comment]
Oh yes, I found that `noinsert` in `set completeopt+=menu,menuone,noinsert` was the key; I was earlier using `set completeopt=menu`.
However, it seems if `completeopt` has `noinsert` and `menu`, but does not have `menuone`, then completion doesn't work at all (even other types). What could be the reason?

**Comment by manasthakur** [comment]
Edit: It works even if `noinsert` is not there. However, `menuone` is a must. I hadn't configured my vim using the recommended settings as the completion plugin I was using earlier had not necessitated `menuone`.

**Comment by lifepillar** [comment]
TL;DR Yes, `menuone` is mandatory. I should reword the documentation.

MUcomplete's way to determine whether a completion method returns some results is to check whether the pop-up menu has become visible. If you remove `menuone`, no menu appears when there is only one completion. But for MUcomplete, it is as if the completion has failed, and tries the next method.

Try for example:

```
xywz xy<tab>
```

Without `menuone`, no completion is performed. Now, try

```
xywz xyaa xy<tab>
```

In this case, the menu appears, so everything works as expected.

**Comment by lifepillar** [comment]
>It works even if `noinsert` is not there

`noinsert` is ignored in manual completion, except for spelling suggestions (see `:h mucomplete-tips`).

**Comment by manasthakur** [comment]
Great! I actually liked the "lightweight" way to check whether a completion failed.
Closing the issue.
